File: python/cases/run_backbone_demo.py
from __future__ import annotations

import sys
from pathlib import Path

SRC = Path(__file__).resolve().parents[1] / "src"
sys.path.insert(0, str(SRC))

from kinetic_mhd_pinn import CoupledState, MODULE_SPECS, nominal_startup_order
from kinetic_mhd_pinn.common.registry import validate_registry


def main() -> None:
    validate_registry()
    state = CoupledState(metadata={"purpose": "architecture smoke test"})
    print("Kinetic MHD-PINN backbone")
    print("Nominal startup order:")
    for module_id in nominal_startup_order():
        spec = next(item for item in MODULE_SPECS if item.id == module_id)
        print(f"  {spec.number:02d}. {spec.name} [{spec.status}]")
        state.publish(module_id, **{f"{module_id}_registered": True})
    print(f"Registered {len(state.history)} module transitions.")
    print("No physics solver was executed; this is an interface-only smoke test.")


if __name__ == "__main__":
    main()

File: matlab/cases/run_backbone_demo.m
function run_backbone_demo()
%RUN_BACKBONE_DEMO Exercise the MATLAB registry and shared-state contract.

casesDir = fileparts(mfilename('fullpath'));
matlabRoot = fileparts(casesDir);
sourceDir = fullfile(matlabRoot, 'src');
addpath(sourceDir);
cleanup = onCleanup(@() rmpath(sourceDir)); %#ok<NASGU>

kinetic_mhd_pinn.common.validate_registry();
specs = kinetic_mhd_pinn.common.registry();
order = kinetic_mhd_pinn.common.nominal_startup_order();
state = kinetic_mhd_pinn.common.CoupledState( ...
    struct('purpose', 'architecture smoke test'));

fprintf('Kinetic MHD-PINN MATLAB backbone\n');
fprintf('Nominal startup order:\n');
for i = 1:numel(order)
    moduleId = order{i};
    index = find(strcmp({specs.id}, moduleId), 1, 'first');
    spec = specs(index);
    fprintf('  %02d. %s [%s]\n', spec.number, spec.name, spec.status);
    state.publish(moduleId, [moduleId '_registered'], true);
end
fprintf('Registered %d module transitions.\n', numel(state.history));
fprintf('No physics solver was executed; this is an interface-only smoke test.\n');
end
